desc/objectives/_coils.py

```diff
@@ -708,6 +716,362 @@ def compute(self, params, constants=None):
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        coilset = self.things[0]
+        eq = self._eq if self._eq_fixed else self.things[1]
+        plasma_grid = self._plasma_grid or LinearGrid(M=eq.M_grid, N=eq.N_grid)
+        coil_grid = self._coil_grid or LinearGrid(N=16)
+        warnif(
+            not np.allclose(plasma_grid.nodes[:, 0], 1),
+            UserWarning,
+            "Plasma/Surface grid includes interior points, should be rho=1.",
+        )
+
+        self._dim_f = coilset.num_coils
+        self._eq_data_keys = ["R", "phi", "Z"]
+
+        eq_profiles = get_profiles(
+            self._eq_data_keys,
+            obj=eq,
+            grid=plasma_grid,
+            has_axis=plasma_grid.axis.size,
+        )
+        eq_transforms = get_transforms(
+            self._eq_data_keys,
+            obj=eq,
+            grid=plasma_grid,
+            has_axis=plasma_grid.axis.size,
+        )
+
+        self._constants = {
+            "coilset": coilset,
+            "coil_grid": coil_grid,
+            "eq_profiles": eq_profiles,
+            "eq_transforms": eq_transforms,
+            "quad_weights": 1.0,
+        }
+
+        if self._eq_fixed:
+            # precompute the equilibrium surface coordinates
+            data = compute_fun(
+                eq,
+                self._eq_data_keys,
+                params=eq.params_dict,
+                transforms=eq_transforms,
+                profiles=eq_profiles,
+            )
+            plasma_coords = rpz2xyz(jnp.array([data["R"], data["phi"], data["Z"]]).T)
+            self._constants["plasma_coords"] = plasma_coords
+
+        if self._normalize:
+            scales = compute_scaling_factors(eq)
+            self._normalization = scales["a"]
+
+        super().build(use_jit=use_jit, verbose=verbose)
+
+    def compute(self, coils_params, eq_params=None, constants=None):
+        """Compute minimum distance between coils and the plasma/surface.
+
+        Parameters
+        ----------
+        coils_params : dict
+            Dictionary of coilset degrees of freedom, eg ``CoilSet.params_dict``
+        eq_params : dict
+            Dictionary of equilibrium or surface degrees of freedom,
+            eg ``Equilibrium.params_dict``
+            Only required if ``self._eq_fixed = False``.
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc.
+            Defaults to self._constants.
+
+        Returns
+        -------
+        f : array of floats
+            Minimum distance from coil to surface for each coil in the coilset.
+
+        """
+        if constants is None:
+            constants = self.constants
+
+        # coil pts; shape(ncoils,coils_grid.num_nodes,3)
+        coils_pts = constants["coilset"]._compute_position(
```

Comment:
Why not use the grid to create a transform here and then compute the coil_points?
This seems more complicated than writing a transform in def compute()

Reply:
`_compute_position` accounts for any possibly stellarator symmetry and NFP, so we need to use that to get the actual positions of every coil in the coilset, not just the unique ones (which is what would happen if you did what you suggest if we have a `CoilSet` with symmetry/NFP).